pkgs/servers/typedb/default.nix

```
@@ -0,0 +1,99 @@
+{ stdenv, lib, openjdk,typedbHome ? "~/.typedb_home", fetchzip, fetchurl}:
+
+let 
+  typedbVersion = "2.1.1";
+  typedbDirLinux = "typedb-all-linux-${typedbVersion}";
+  typedbDirMac = "typedb-all-mac-${typedbVersion}";
+  typedbDirWindows = "typedb-all-windows-${typedbVersion}";
+  typedbDir = if stdenv.hostPlatform.isWindows then typedbDirWindows
+              else if stdenv.isDarwin          then typedbDirMac
+                                               else typedbDirLinux;
+  linuxSrc = builtins.fetchTarball {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+    sha256 = "15nwm2dr68p67c2xcqigs66gd679j1zr72gqv7qxgvflwyvyz8fb";
+  };
+  linuxSrcUrl = fetchurl {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+    sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
+  };
+  windowsSrc = fetchzip {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-windows-2.1.1.zip";
+    sha256 = "0vd66gfshkg697z07nhy957mwqzlli4r4pmn67hx58n9mkg024kq";
+  };
+  macSrc = fetchzip {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-mac-2.1.1.zip";
+    sha256 = "16hlfy6kh2rnvcralz206q13mghb0rv8wazpg6q3h324p5rdys54";
+  };
+  srcFolder = if stdenv.hostPlatform.isWindows then windowsSrc
+              else if stdenv.isDarwin          then macSrc
+                                               else linuxSrcUrl ;
+  javaPatch = ''
+        20c20
+        < JAVA_BIN=java
+        ---
+        > JAVA_BIN=${openjdk}/bin/java
+        '';
+
+in
+stdenv.mkDerivation rec {
+  pname = "typedb";
+  version = typedbVersion;
+
+  src = srcFolder;
+    
+  phases = [ "installPhase" "installCheckPhase" ];
```

This is the culprit. `unpackPhase` was missing, but better is:

```suggestion
  dontBuild = true;
```

---

ah, ok this makes sense now thanks

---

a question regarding this:
if i `fetchurl` why is the tar.gz file automatically unpacked?
is this related to the unpackPhase?

---

Exactly. `tar` is part of stdenv, so its unpackPhase behavior is also in `stdenv`

https://github.com/NixOS/nixpkgs/blob/5cc08e40f699674505e80f9add379b1bb9133da3/pkgs/stdenv/generic/setup.sh#L813

`nativeBuildInputs = [ unzip ];` would cause the following setup hook to run

https://github.com/NixOS/nixpkgs/blob/bd1630ed0a688bd02129d3a9e687f5ac89dc9d49/pkgs/tools/archivers/unzip/setup-hook.sh#L1

but you don't need that because the fetcher already unpacks it beforehand.

In fact, `fetchzip` is implemented with `unpackFile`, backed by the same hook mechanism, so it supports tarballs too. Most packages use `fetchurl` though.